maths/krishnamurthy_number_2.py

```diff
@@ -0,0 +1,81 @@
+import doctest
+
+"""
+A Krishnamurthy number: A number that's equal to the sum of the factorials of its digits
+
+For instance, 145 = 1! + 4! + 5! ; 40585 = 4! + 0! + 5! + 8! + 5!, but 78 != 7! + 8!
+"""
+
+
+def factorial(number: int) -> int:
+    # Function to calculate the factorial of a number
+    # n! = n*(n-1)*(n-2)*...*2*1, that is the definition of the factorial of n (n!)
+
+    """
+    >>> factorial(7)
+    5040
+    >>> factorial("some characters") #like '1' or 'a'
+    Traceback (most recent call last):
+      File "<stdin1>", line 1, in <module>
+    TypeError: Only integers are allowed
+    >>> factorial(1)
+    1
+    >>> factorial(-4)
+    Traceback (most recent call last):
+      File "<stdin>", line 1, in <module>
+    ValueError: Only positive integers are allowed
+    >>> factorial(0)
+    1
+    >>> factorial(4)
+    24
+    """
+    if not isinstance(number, int):  # If n is not an integer
```

Comment:
We'll I don't think that you need to provide comments on these codes (for negative number, instance of int and other below),which is very simple to understand.